common/test/java/com/couchbase/lite/BaseDbTest.java

```diff
@@ -1655,14 +1665,38 @@ protected final void verifyDocument(DictionaryInterface doc) {
         verifyBlob(doc.getBlob("doc-29"));
     }
 
+    protected Database openDatabase() throws CouchbaseLiteException { return verifyDb(createDb("test_db")); }
+
     protected final void reopenBaseTestDb() throws CouchbaseLiteException { baseTestDb = reopenDb(baseTestDb); }
 
     protected final void recreateBastTestDb() throws CouchbaseLiteException { baseTestDb = recreateDb(baseTestDb); }
 
+    protected Database duplicateBaseTestDb() throws CouchbaseLiteException { return verifyDb(duplicateDb(baseTestDb)); }
+
+
     // Some JSON encoding will promote a Float to a Double.
     protected final Float demoteToFloat(Object val) {
         if (val instanceof Float) { return (Float) val; }
         if (val instanceof Double) { return ((Double) val).floatValue(); }
         throw new IllegalArgumentException("expected a floating point value");
     }
+    private Database verifyDb(Database db) {
+        try {
+            assertNotNull(db);
+            assertTrue(new File(db.getPath()).getCanonicalPath().endsWith(C4Database.DB_EXTENSION));
+
+            return db;
+        }
+        catch (IOException e) {
+            deleteDb(db);
+            throw new AssertionError("Unable to get db path", e);
+        }
+        catch (AssertionError e) {
```

Comment:
This shouldn't be a `catch` block, it should be a `finally` block

Reply:
it shouldn't be a finally block since we should only delete the database if there is an error in verifying the database

Reply:
Ahhh!!!! Gottcha.  This isn't a test, its a utility method.
In that case, I don't think it should delete the database at all.  It should return true or false and the db creator should be responsible for doing the right thing...

Reply:
Or.... if that would break a lot of legacy code, just rename it "verifyOrDeleteDb"...